From 65f560005f309a24090bdaa43107692968208483 Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Fri, 6 Nov 2020 21:40:29 -0800
Subject: [PATCH 1/1] build: Set ro.apex.updatable=false in product properties

Devices using prebuilt vendor images may set ro.apex.updatable=true in
/vendor/build.prop, which causes boot failures because our APEXs are not
built as updatable and thus never get mounted.

Because init reads /product/build.prop *after* /vendor/build.prop and
overrides read-only properties during early boot prop reading, we can
set ro.apex.updatable=false explicitly in product properties to fix this
issue.
---
 config/common.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/config/common.mk b/config/common.mk
index df4d638f..79a341c9 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -210,6 +210,9 @@ endif
 
 # Flatten APEXs for performance
 OVERRIDE_TARGET_FLATTEN_APEX := true
+# This needs to be specified explicitly to override ro.apex.updatable=true from
+# # prebuilt vendors, as init reads /product/build.prop after /vendor/build.prop
+PRODUCT_PRODUCT_PROPERTIES += ro.apex.updatable=false
 
 # Repainter integration
 PRODUCT_PACKAGES += \
-- 
2.36.1

